janitor/functions/pivot.py

```diff
@@ -206,60 +236,84 @@ def pivot_wider(
+        ...       {'dep': 6.1, 'step': 1, 'a': 22, 'b': 19},
+        ...       {'dep': 6.1, 'step': 2, 'a': 18, 'b': 29}]
+        >>> df = pd.DataFrame(df)
+        >>> df
+           dep  step   a   b
+        0  5.5     1  20  30
+        1  5.5     2  25  37
+        2  6.1     1  22  19
+        3  6.1     2  18  29
+
+    Pivot and flatten columns:
+
+        >>> df.pivot_wider(
+        ...     index = "dep",
+        ...     names_from = "step",
+        ... )
+           dep  a_1  a_2  b_1  b_2
+        0  5.5   20   25   30   37
+        1  6.1   22   18   19   29
+
+    Change the order of the column labels:
+
+        >>> df.pivot_wider(
+        ...     index = "dep",
+        ...     names_from = "step",
+        ...     levels_order = ["step", None],
+        ... )
+           dep  1_a  2_a  1_b  2_b
+        0  5.5   20   25   30   37
+        1  6.1   22   18   19   29
+
+    Modify columns with `names_sep`:
+
+        >>> df.pivot_wider(
+        ...     index = "dep",
+        ...     names_from = "step",
+        ...     names_sep = "",
+        ... )
+           dep  a1  a2  b1  b2
+        0  5.5  20  25  30  37
+        1  6.1  22  18  19  29
+
+    Modify columns with `names_glue`:
+
+        >>> df.pivot_wider(
+        ...     index = "dep",
+        ...     names_from = "step",
+        ...     names_sep = None,
+        ...     names_glue = lambda col: f"{col[0]}_step{col[1]}",
+        ... )
+           dep  a_step1  a_step2  b_step1  b_step2
+        0  5.5       20       25       30       37
+        1  6.1       22       18       19       29
 
-        df = pd.DataFrame(...)
-
-        df = jn.pivot_wider(
-            df = df,
-            index = [column1, column2, ...],
-            names_from = [column3, column4, ...],
-            value_from = [column5, column6, ...],
-            levels_order = None/list,
-            flatten_levels = True/False,
-            names_sep='_',
-            names_glue= Callable
-
-        )
-    ```
-
-    Method chaining syntax:
-
-    ```python
-        df = (
-            pd.DataFrame(...)
-            .pivot_wider(
-                index = [column1, column2, ...],
-                names_from = [column3, column4, ...],
-                value_from = [column5, column6, ...],
-                levels_order = None/list,
-                flatten_levels = True/False,
-                names_sep='_',
-                names_glue= Callable
-                )
-        )
-    ```
 
     :param df: A pandas DataFrame.
     :param index: Name(s) of columns to use as identifier variables.
         It should be either a single column name, or a list of column names.
-        The `janitor.select_columns` syntax is supported here,
-        allowing for flexible and dynamic column selection.
+        The [`select_columns`][janitor.functions.select_columns.select_columns]
+        is supported here, allowing for flexible and dynamic column selection.
```

Comment:
```suggestion
        syntax is supported here, allowing for flexible and dynamic column selection.
```

Reply:
@samukweku, rather than repeat ourselves here, what do you think of writing the comment above the `:params` section that the select columns syntax is supported in `index`, `names_from`, and `value_from`? Doing so would shorten the docstring and make it easier to read.